src/export/StructureDefinitionExporter.ts

```diff
@@ -154,6 +155,16 @@ export class StructureDefinitionExporter implements Fishable {
             } else {
               structDef.setInstancePropertyByPath(rule.caretPath, rule.value, this);
             }
+          } else if (rule instanceof ObeysRule) {
+            const invariant = this.tank.getAllInvariants().find(i => i.name === rule.invariant);
```

Comment:
Does it make sense to use wrap this into the `fish` method on `FSHTank`? So we would get this like:
```
const invariant = this.tank.fish(i.name, Type.Invariant)
```

I am currently thinking about how to do this with `Mixins` and I'd want to be consistent. On the one hand since `Invariant`s should only be referred to by their name, maybe putting them in `fish` is overkill. On the other hand, maybe it is more consistent to always be finding stuff with `fish`.

Reply:
Ah that's a good point! I hadn't thought of that, but I kind of like the idea of adding it into `fish`. I think I personally like the consistency and would be okay with having it look through the list only by name, but I also haven't touched this function a whole lot, so I'm open to others' opinions.

Reply:
I like to `fish` (metaphorically).  So I'm in favor of adding it to the fisher.

Reply:
Should I add the ability to fish by `id` as well as `name`? Presumably we'd only be using `name` right now, but I can add the other for consistency if we want.

Reply:
I don't think invariants have an id, do they?  Don't they only have a name?
<img width="458" alt="image" src="https://user-images.githubusercontent.com/2278253/75574816-f5602080-5a2c-11ea-9557-eeddb9477a4f.png">
Or am I misunderstanding your question?

Reply:
Ah no, you are correct. However, typescript gets mad at all the other places we are accessing `id` on the objects we've `fish`ed for because, until now, all the types that `fish` returns have an `id`. 

@cmoesel stopped by as I was writing this, and I'm going to add a read-only `id` property onto Invariant to make sure we can still access `.id` on anything we've `fish`ed for. I won't add any ability to `fish` by that `id` though.

Reply:
The last commit should cover this!